lms/static/scripts/frontend_apps/services/test/index-test.js

```diff
@@ -0,0 +1,60 @@
+import { mount } from 'enzyme';
+import { createElement } from 'preact';
+
+import { Services, useService, withServices } from '../index';
+
+class DummyService {}
+
+describe('services/index', () => {
+  describe('useService', () => {
+    it('returns matching registered service', () => {
+      const services = new Map([[DummyService, new DummyService()]]);
+
+      let dataService;
+      const Widget = () => {
+        dataService = useService(DummyService);
+        return null;
+      };
+      mount(
+        <Services.Provider value={services}>
+          <Widget />
+        </Services.Provider>
+      );
+
+      assert.equal(dataService, services.get(DummyService));
+    });
+
+    it('throws if service is not registered', () => {
+      const TestComponent = () => {
+        useService(DummyService);
+        return null;
+      };
+
+      assert.throws(() => {
+        mount(
+          <Services.Provider value={new Map()}>
```

Comment:
I think the `value` prop is required for a `*.Provider` component. If there was no `Services.Provider` in the tree then that would function exactly the same as this code, because the default value is an empty map. This test is just being explicit.